### DISCUSSION TITLE
Possible bug with additional/watched_files or are we doing it wrong?

### DISCUSSION THREAD

**Discussion by danielnolan** [shakacode/shakapacker#164] [Q&A] [answered]
So we have added some paths to the `additional_paths` like `app/assets/stylesheets`. We only import like one or two stylesheets from the root of this directory like `stylesheets/variables.scss`. The way the `additional_paths` are added to `watched_files` like `app/assets/stylesheets/**/*` if any file changes in stylesheets or any of it's child directories a webpack compile is triggered cause the watched file digest changes. How would we go about being able to import one or two files from the root of stylesheets without having stylesheets and all of it's children becoming `watched_files`? I would expect there to be a way to only have a compile triggered if the actual files we are importing change. webpacker used to have a separate watched_files config, maybe that should be brought back and the additional_paths config shouldn't be a one size fits all config? Are we doing it wrong or is this something that has been overlooked with the additional_paths config?

**Comment by Judahmeek** [comment, marked as answer]
You should be able add individual files to additional_paths, rather than just entire directories.

**Reply by danielnolan** [reply]
@Judahmeek so [this line](https://github.com/shakacode/shakapacker/blob/6dcbee5beb95878a96f7578f16ba5dbf891b325f/lib/shakapacker/base_strategy.rb#L17) allows for individual files? We tried adding a single file and made changes to it and it wouldn't trigger a compile.

**Reply by Judahmeek** [reply]
@danielnolan yes, it does.

That said, I've also confirmed that making changes to an individual file listed in `additional_paths` does not trigger a recompile, so I will investigate this further.

**Reply by Judahmeek** [reply]
@danielnolan After further research & experimentation, I think you might find the simplest solution is an empty `additional_paths`.

According to my tests with https://github.com/shakacode/shakapacker/tree/judahmeek/additional-paths-example, `additional_paths` isn't necessary for importing individual files from outside the configured `source_path` or for recompilation when changes are made to those individual files.

That's as long as you're using relative paths for the import statement.

If you're using module resolution for your import statement, then you might try an empty `additional_paths` & a modified `resolve.modules` rule in your `/config/webpack/...`

Btw, I wasn't able to trigger a recompilation unless the file in `additional_paths` was also imported by source code while using `bin/shakapacker-dev-server` or `bin/shakapacker --watch`, so would I be correct to assume that you have been using `compile: true` for `development` in your Shakapacker configuration file?

That's the only way I can imagine changes to files you haven't imported, but have included in your `additional_paths` could be triggering recompilation.
